flytekit/bin/entrypoint.py

```diff
@@ -177,6 +177,13 @@ def _dispatch_execute(
     for k, v in output_file_dict.items():
         utils.write_proto_to_file(v.to_flyte_idl(), os.path.join(ctx.execution_state.engine_dir, k))
 
+    # make sure an event loop exists for data persistence step
+    try:
+        asyncio.get_event_loop()
+    except Exception:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
```

Comment:
Should we specify which exception we might have?

Reply:
ah yes this should be `RuntimeError`